tqa-web-app/src/app/state/store/resources/adminauth/adminauth.effects.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import { Actions, createEffect, ofType } from "@ngrx/effects";
+import { Action, Store } from "@ngrx/store";
+import { of } from "rxjs";
+import { exhaustMap, map, mergeMap, tap } from "rxjs/operators";
+import { AdminV2 } from "src/app/dtos/v2/admin-auth.dto.v2";
+import { AdminAuthService } from "src/app/services/adminauth.service";
+import { AppState } from "../../app.state";
+import { adminAutoLogin, adminAutoLogout, adminLoginStart, adminLoginSuccess } from "./adminauth.actions";
+
+@Injectable()
+export class AdminAuthEffects{
+    constructor(
+        private actions$:Actions, 
+        private adminauthService: AdminAuthService, 
+        private store: Store<AppState>,
+        private router: Router,
+        ){}
+
+    adminLogin$ = createEffect(()=>{
+        return this.actions$.pipe(
+            ofType(adminLoginStart), 
+            exhaustMap((action)=>{
+                return this.adminauthService.login(action.email, action.password).pipe(
+                    map((data)=>{
+                        //Formatting the data and pass it to the action adminLoginSuccess
+                        const admin = this.adminauthService.formatAdmin(data);
+                        //Set the user in local storage
+                        this.adminauthService.setAdminInLocalStorage(admin);
+                        return adminLoginSuccess({admin});
+                    })
+                )
+            })
+        );
+    })
+
+    loginRedirect$ = createEffect(
+        () => {
+          return this.actions$.pipe(
+            ofType(adminLoginSuccess),
+            tap((action) => {
+              this.router.navigate(['/']);
+            })
+          );
+        },
+        { dispatch: false }
+      );
```

Comment:
I would make sure the formatting of this matches across all effects.  Based on how we have done the rest, move the () => { up a line for all methods here and below.

Reply:
amended